clippy_utils/src/macros.rs

```diff
@@ -757,6 +772,20 @@ pub struct FormatArg<'tcx> {
     pub span: Span,
 }
 
+impl<'tcx> FormatArg<'tcx> {
+    /// Span of the `:` and format specifiers
+    ///
+    /// ```ignore
+    /// format!("{:.}"), format!("{foo:.}")
+    ///           ^^                  ^^
+    /// ```
+    pub fn format_span(&self) -> Span {
+        let base = self.span.data();
+
+        Span::new(self.param.span.hi(), base.hi - BytePos(1), base.ctxt, base.parent)
```

Comment:
All `BytePos` usages to build spans should come with a comment, why it is always safe to use it. This includes weird formatting. I assume that the `base` span includes the final `}`? Then you should add a comment that it does and that the `BytePos` removes that.